### PR TITLE
feat(frontend): Add view mode toggle (old/diff/new) to file changes viewer

### PR DESCRIPTION
**Pull request by VascoSch92**
## Summary of PR

**Problem**
The changes tab only offered a diff view for each file. When reviewing markdown files, the raw diff was hard to read — there was no way to see the rendered result. More generally, there was no way to inspect the original or modified version of a file in isolation. 

**Solution**               
Added three icon-only view mode buttons to each file's header in the changes tab:                              
- Old (LuHistory) — shows the original version of the file                                                                                                                                                                             
- Diff (LuGitCompareArrows) — the existing side-by-side diff (default, unchanged behavior)                                                                                                                                             
- New (LuFileCheck) — shows the modified version of the file                                                                                                                                                                           

For markdown files (.md), the old/new views render the content using the existing MarkdownRenderer component with full GFM support (headings, links, tables, code blocks, etc.). For all other files, old/new views use a read-only Monaco Editor with syntax highlighting.                                                                                                                                                                                                

Icons were chosen deliberately to avoid any translatable strings — no i18n changes needed.                                                                                                                                             
   
**Code quality improvements made along the way:**           
- Extracted SHARED_EDITOR_OPTIONS, VIEW_MODES, and beforeMount as module-level constants (no per-render recreation)
- Extracted EditorContainer to deduplicate the wrapper div shared by Editor and DiffEditor                                                                                                                                             
- Replaced nested ternary JSX with a renderContent() helper using clear if/return blocks  
- Avoided redundant getLanguageFromPath calls 

## demo 

before the change
Screenshot 2026-03-21 at 20 26 36

diff
Screenshot 2026-03-21 at 20 26 52

after the change
Screenshot 2026-03-21 at 20 27 05

if we have a Markdown, we render it
Screenshot 2026-03-21 at 20 27 21

## Change Type

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [x] Refactor
- [ ] Other (dependency update, docs, typo fixes, etc.)

## Checklist

- [x] I have read and reviewed the code and I understand what the code is doing.
- [x] I have tested the code to the best of my ability and ensured it works as expected.

## Fixes

Resolves #(issue)

## Release Notes

- [x] Include this change in the Release Notes.

### CODE REVIEW

File: frontend/src/components/features/diff-viewer/file-diff-viewer.tsx [hunk 1]
```diff
@@ -1,20 +1,49 @@
...
+} from "react-icons/lu";
 import { IconType } from "react-icons/lib";
 import { GitChangeStatus } from "#/api/open-hands.types";
 import { getLanguageFromPath } from "#/utils/get-language-from-path";
 import { cn } from "#/utils/utils";
 import ChevronUp from "#/icons/chveron-up.svg?react";
 import { useUnifiedGitDiff } from "#/hooks/query/use-unified-git-diff";
+import { MarkdownRenderer } from "#/components/features/markdown/markdown-renderer";
 
-interface LoadingSpinnerProps {
```
Comment (hieptl) [inline frontend/src/components/features/diff-viewer/file-diff-viewer.tsx:12]: **suggestion (blocking):** Could we please add this back when you have a moment?

Thank you! 🙏

File: frontend/src/components/features/diff-viewer/file-diff-viewer.tsx [hunk 2]
```diff
@@ -1,20 +1,49 @@
...
+  A: LuFilePlus,
+  D: LuFileMinus,
+  M: LuFileDiff,
+  R: "Renamed",
+  U: "Untracked",
+};
 
 // TODO: Move out of this file and replace the current spinner with this one
-function LoadingSpinner({ className }: LoadingSpinnerProps) {
+function LoadingSpinner({ className }: { className?: string }) {
```
Comment (hieptl) [inline frontend/src/components/features/diff-viewer/file-diff-viewer.tsx]: **suggestion (blocking):** Would it be possible to move this component into a separate file for better organization? Alternatively, if feasible, we could consider removing it and reusing the existing LoadingSpinner component.

Thank you! 🙏

File: frontend/src/components/features/diff-viewer/file-diff-viewer.tsx [hunk 3]
```diff
@@ -29,14 +58,45 @@ function LoadingSpinner({ className }: LoadingSpinnerProps) {
...
+      "diffEditor.insertedLineBackground": "#003f00AA",
+      "diffEditor.removedLineBackground": "#5a0000AA",
+      "diffEditor.border": "#444444",
+      "editorUnnecessaryCode.border": "#00000000",
+      "editorUnnecessaryCode.opacity": "#00000077",
+    },
+  });
 };
 
+function EditorContainer({
```
Comment (hieptl) [inline frontend/src/components/features/diff-viewer/file-diff-viewer.tsx]: **suggestion (blocking):** Could you please move this component into a separate file and add tests for it?

Thank you! 🙏

File: frontend/src/components/features/diff-viewer/file-diff-viewer.tsx [hunk 4]
```diff
@@ -70,69 +131,106 @@ export function FileDiffViewer({ path, type }: FileDiffViewerProps) {
...
-  let statusIcon: React.ReactNode;
-  if (typeof status === "string") {
-    statusIcon = <span>{status}</span>;
-  } else {
-    const StatusIcon = status; // now it's recognized as a component
-    statusIcon = <StatusIcon className="w-5 h-5" />;
-  }
+  const statusIcon =
+    typeof status === "string" ? (
+      <span>{status}</span>
```
Comment (hieptl) [inline frontend/src/components/features/diff-viewer/file-diff-viewer.tsx]: **suggestion (blocking):** We should use Typography.Text from the /frontend/src/ui folder. Thank you! 🙏

File: frontend/src/components/features/diff-viewer/file-diff-viewer.tsx [hunk 5]
```diff
@@ -144,9 +242,31 @@ export function FileDiffViewer({ path, type }: FileDiffViewerProps) {
         onClick={() => setIsCollapsed((prev) => !prev)}
       >
         <span className="text-sm w-full text-content flex items-center gap-2">
-          {isFetchingData && <LoadingSpinner className="w-5 h-5" />}
-          {!isFetchingData && statusIcon}
+          {isFetchingData ? <LoadingSpinner className="w-5 h-5" /> : statusIcon}
           <strong className="w-full truncate">{filePath}</strong>
+          {!isCollapsed && (
+            <span
```
Comment (hieptl) [inline frontend/src/components/features/diff-viewer/file-diff-viewer.tsx:218]: suggestion (blocking): We should use Typography.Text from the /frontend/src/ui folder. Thank you! 🙏

File: frontend/src/components/features/diff-viewer/file-diff-viewer.tsx [hunk 6]
```diff
@@ -29,14 +58,45 @@ function LoadingSpinner({ className }: LoadingSpinnerProps) {
...
+  height,
+  children,
+}: {
+  height: number;
+  children: React.ReactNode;
+}) {
+  return (
+    <div
+      className="w-full border border-neutral-600 overflow-hidden"
+      style={{ height: `${height}px` }}
```
Comment (hieptl) [inline frontend/src/components/features/diff-viewer/file-diff-viewer.tsx]: **suggestion (blocking):** It would be best to avoid using inline styles. Thank you! 🙏

File: frontend/src/components/features/diff-viewer/file-diff-viewer.tsx (hunk 1 above)
Comment (VascoSch92) [reply frontend/src/components/features/diff-viewer/file-diff-viewer.tsx:12]: The LoadingSpinnerProps interface is now defined as a named, exported interface in the newly extracted loading-spinner.tsx file. It cannot be re-imported into file-diff-viewer.tsx without usage as eslint flags it as an unused import.

File: frontend/src/components/features/diff-viewer/file-diff-viewer.tsx (hunk 2 above)
Comment (VascoSch92) [reply frontend/src/components/features/diff-viewer/file-diff-viewer.tsx]: Extracted LoadingSpinner into frontend/src/components/features/diff-viewer/loading-spinner.tsx. The existing shared LoadingSpinner (src/components/shared/loading-spinner.tsx) has a different API (requires size prop, uses an SVG asset).

File: frontend/src/components/features/diff-viewer/file-diff-viewer.tsx (hunk 3 above)
Comment (VascoSch92) [reply frontend/src/components/features/diff-viewer/file-diff-viewer.tsx]: Extracted into frontend/src/components/features/diff-viewer/editor-container.tsx with 4 tests in editor-container.test.tsx covering: rendering children, test id presence, CSS variable for height, and border/overflow class application.

File: frontend/src/components/features/diff-viewer/file-diff-viewer.tsx (hunk 6 above)
Comment (VascoSch92) [reply frontend/src/components/features/diff-viewer/file-diff-viewer.tsx]: Replaced with a CSS custom property approach: style={{ "--editor-height": \${height}px` }}combined with the Tailwind classh-[var(--editor-height)]`. The dynamic height still requires a style attribute for the variable,
  but layout is driven by Tailwind.

File: frontend/src/components/features/diff-viewer/file-diff-viewer.tsx (hunk 5 above)
Comment (VascoSch92) [reply frontend/src/components/features/diff-viewer/file-diff-viewer.tsx:218]: This uses an onClick handler for stopPropagation(), and Typography.Text doesn't accept event handler props. Keeping as a plain to preserve functionality.

**Comment by VascoSch92** [comment]
@hieptl Done :-)
